op-e2e/actions/kromampt_fork_test.go

```diff
@@ -0,0 +1,182 @@
+package actions
+
+import (
+	"context"
+	"testing"
+
+	oppredeploys "github.com/kroma-network/kroma/op-bindings/predeploys"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/kroma-network/kroma/kroma-bindings/bindings"
+	"github.com/kroma-network/kroma/kroma-bindings/predeploys"
+	"github.com/kroma-network/kroma/kroma-chain-ops/genesis"
+)
+
+var (
+	l1BlockMPTCodeHash           = common.HexToHash("0xc88a313aa75dc4fbf0b6850d9f9ae41e04243b7008cf3eadb29256d4a71c1dfd")
+	baseFeeVaultMPTCodeHash      = common.HexToHash("0xc0ccbda46b89a834c65c871fb0ccb93f02c60268d5560a8e1b13722979bb38dd")
+	sequencerFeeVaultMPTCodeHash = common.HexToHash("0xc0ccbda46b89a834c65c871fb0ccb93f02c60268d5560a8e1b13722979bb38dd")
+	l1FeeVaultMPTCodeHash        = common.HexToHash("0xf96923a0115890fc344bc26402cbdc062ff2d6f63317b40dd7f284f2b37c6d64")
+	gasPriceOracleMPTCodeHash    = common.HexToHash("0x3474b859129195c11a4b1aa14af951a2b3f83c7841aa5e73e67d55bd3379220c")
+)
+
+func TestKromaMPTNetworkUpgradeTransactions(gt *testing.T) {
+	t := NewDefaultTesting(gt)
+	dp := e2eutils.MakeDeployParams(t, defaultRollupTestParams)
+	genesisBlock := hexutil.Uint64(0)
+	kromaMPTOffset := hexutil.Uint64(4)
+
+	dp.DeployConfig.L1CancunTimeOffset = &genesisBlock // can be removed once Cancun on L1 is the default
+
+	// Activate all forks at genesis, and schedule KromaMPT the block after
+	dp.DeployConfig.L2GenesisRegolithTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisCanyonTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisDeltaTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisEcotoneTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisKromaMPTTimeOffset = &kromaMPTOffset
+	require.NoError(t, dp.DeployConfig.Check(), "must have valid config")
+
+	sd := e2eutils.Setup(t, dp, defaultAlloc)
+	log := testlog.Logger(t, log.LevelDebug)
+	_, _, _, sequencer, engine, verifier, _, _ := setupReorgTestActors(t, dp, sd, log)
+	ethCl := engine.EthClient()
+
+	// start op-nodes
+	sequencer.ActL2PipelineFull(t)
+	verifier.ActL2PipelineFull(t)
+
+	// Get gas price from oracle
+	gasPriceOracle, err := bindings.NewGasPriceOracleCaller(predeploys.GasPriceOracleAddr, ethCl)
+	require.NoError(t, err)
+
+	// Get fee vault recipients
+	protocolFeeVault, err := bindings.NewProtocolVaultCaller(predeploys.ProtocolVaultAddr, ethCl)
+	require.NoError(t, err)
+	protocolFeeRecipient, err := protocolFeeVault.RECIPIENT(nil)
+	require.NoError(t, err)
```

Comment:
How about adding the checks that recipient address is not a zero address, and matches with the deploy config?

Reply:
Could you add it to l1FeeVault also?